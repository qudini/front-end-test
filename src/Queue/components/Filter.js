// components/Filter.js
import React from 'react';
import styled from 'styled-components';

const FilterWrapper = styled.div`
  padding-bottom: 10px;
  width: 342px;
`;

const FilterLabel = styled.label`
  display: block;
  margin-bottom: 10px;
  font-weight: bold;
  color: darkslategrey;
`;

const FilterInput = styled.input`
  width: 100%;
  padding: 8px;
  box-sizing: border-box;
`;

const Filter = ({ filterValue, onFilterChange }) => {
    return (
        <FilterWrapper>
            <FilterLabel htmlFor="filterInput">Filter</FilterLabel>
            <FilterInput
                type="text"
                id="filterInput"
                placeholder="Filter by name"
                value={filterValue}
                onChange={(e) => onFilterChange(e.target.value)}
            />
        </FilterWrapper>
    );
};

export default Filter;
