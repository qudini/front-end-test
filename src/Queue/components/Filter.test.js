import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Filter from './Filter';
import '@testing-library/jest-dom';
import { axe } from 'jest-axe';

describe('Filter Component', () => {
    it('renders the Filter component', () => {
        render(<Filter filterValue="" onFilterChange={() => {}} />);
        expect(screen.getByLabelText(/Filter/)).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<Filter filterValue="" onFilterChange={() => {}} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('is accessible', async () => {
        const { container } = render(<Filter filterValue="" onFilterChange={() => {}} />);
        const results = await axe(container);
        expect(results).toHaveNoViolations();
    });

    it('invokes onFilterChange callback on input change', async () => {
        const mockCallback = jest.fn();
        render(<Filter filterValue="" onFilterChange={mockCallback}/>);
        const input = screen.getByLabelText(/Filter/);

        await userEvent.type(input, 't');
        expect(mockCallback).toHaveBeenCalledWith('t');
    });
});
