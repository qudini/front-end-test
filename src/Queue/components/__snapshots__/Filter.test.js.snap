// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  padding-bottom: 10px;
  width: 342px;
}

.c1 {
  display: block;
  margin-bottom: 10px;
  font-weight: bold;
  color: darkslategrey;
}

.c2 {
  width: 100%;
  padding: 8px;
  box-sizing: border-box;
}

<div
    class="c0"
  >
    <label
      class="c1"
      for="filterInput"
    >
      Filter
    </label>
    <input
      class="c2"
      id="filterInput"
      placeholder="Filter by name"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
